ale_linters/rust/cargo.vim

```diff
@@ -82,6 +86,7 @@ function! ale_linters#rust#cargo#GetCommand(buffer, version) abort
     \   . (l:use_all_targets ? ' --all-targets' : '')
     \   . (l:use_examples ? ' --examples' : '')
     \   . (l:use_tests ? ' --tests' : '')
+    \   . (l:use_target_dir ? (' --target-dir ' . l:target_dir) : '')
```

Comment:
You need to quote paths with the `ale#Escape` function, which has cross-platform support.

```suggestion
    \   . (l:use_target_dir ? (' --target-dir ' . ale#Escape(l:target_dir)) : '')
```

Have a look at the `command_callback` tests documented in `:help ale-dev`. You can test the strings used for any command, and the versions we check. There will be number of tests for cargo already, so you can look at the current examples and add another test case.